### PR TITLE
Fix secret to backupentry and backupbucket extension resource mapper

### LINKED ISSUE #2475: Wrong usage of client.MatchingFields
**How to categorize this issue?**

/area quality
/kind bug
/priority normal

**What happened**:
Currently `secretToBackupBucketMapper` and `secretToBackupEntryMapper` both have usage wrong usage of `client.MatchingFields`.

https://github.com/gardener/gardener/blob/973993fdea1b0cfd843c9795f65fe4d9ad898fa2/extensions/pkg/controller/backupbucket/mapper.go#L61-L64

https://github.com/gardener/gardener/blob/973993fdea1b0cfd843c9795f65fe4d9ad898fa2/extensions/pkg/controller/backupentry/mapper.go#L47-L50

Both mappers will always return nil as the `List` invocation always returns non-nil err `field label not supported: spec.secretRef.namespace`.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
You can put a log entry in the mapper to see that the error is always non-nil. Or you can use the follow `main.go` to reproduce the issue:

```golang
func main() {
	scheme := runtime.NewScheme()
	extensionsv1alpha1.AddToScheme(scheme)

	cl, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme})
	if err != nil {
		panic(err)
	}

	backupBucketList := &extensionsv1alpha1.BackupBucketList{}
	if err := cl.List(context.TODO(), backupBucketList,
		client.MatchingFields{"spec.secretRef.name": "foo"}, client.MatchingFields{"spec.secretRef.namespace": "bar"}); err != nil {
		panic(err)
	}

	fmt.Printf("%+v", backupBucketList)
}
```

```
$ go run main.go
panic: field label not supported: spec.secretRef.namespace

goroutine 1 [running]:
main.main()
	/Users/me/git/example/main.go:25 +0x32b
exit status 2
```

**Anything else we need to know?**:

**Environment**:

- Gardener version: v1.6.2
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by swapnilgm**
**How to categorize this PR?**

/area backup
/kind bug
/priority normal

**What this PR does / why we need it**:
This fixes the secret to backupEntry and backupBuckets resource mappers.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/2475

**Special notes for your reviewer**:
To test the PR, you have to revendor provider extension and you can put addition log message to verify that mapper function works as expected.

**Release note**:

```improvement operator
Fix secret to backupBucket and backupEntry extension resource mapper.
```

### CODE REVIEW

File: extensions/pkg/controller/backupbucket/mapper.go [hunk 1]
```diff
@@ -59,20 +59,21 @@ func (m *secretToBackupBucketMapper) Map(obj handler.MapObject) []reconcile.Requ
 	}
 
 	backupBucketList := &extensionsv1alpha1.BackupBucketList{}
-	if err := m.client.List(context.TODO(), backupBucketList, client.MatchingFields{"spec.secretRef.name": secret.Name}, client.MatchingFields{"spec.secretRef.namespace": secret.Namespace}); err != nil {
+	if err := m.client.List(context.TODO(), backupBucketList, &client.ListOptions{}); err != nil {
```
Comment (ialidzhikov) [inline extensions/pkg/controller/backupbucket/mapper.go]:
```suggestion
	if err := m.client.List(context.TODO(), backupBucketList); err != nil {
```

File: extensions/pkg/controller/backupentry/mapper.go [hunk 2]
```diff
@@ -45,21 +45,23 @@ func (m *secretToBackupEntryMapper) Map(obj handler.MapObject) []reconcile.Reque
 	}
 
 	backupEntryList := &extensionsv1alpha1.BackupEntryList{}
-	if err := m.client.List(context.TODO(), backupEntryList, client.MatchingFields{"spec.secretRef.name": secret.Name}, client.MatchingFields{"spec.secretRef.namespace": secret.Namespace}); err != nil {
+	if err := m.client.List(context.TODO(), backupEntryList, &client.ListOptions{}); err != nil {
```
Comment (ialidzhikov) [inline extensions/pkg/controller/backupentry/mapper.go]:
```suggestion
	if err := m.client.List(context.TODO(), backupEntryList); err != nil {
```

**Review by timuthy — APPROVED** [review]
/lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**